lib/common/src/Url.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace AmpProject;
+
+/**
+ * Helper class to work with URLs.
+ *
+ * @package ampproject/common
+ */
+final class Url
+{
+
+    /**
+     * Default URL parts to use when constructing an absolute URL out of a relative one.
+     *
+     * @var string[]
+     */
+    const URL_DEFAULT_PARTS = [
+        'scheme'   => 'https',
+        'host'     => 'example.com',
```

Comment:
This doesn't seem right? Shouldn't it be using the host from `home_url()`?

cf. `amp_get_current_url()`

Reply:
`home_url()` doesn't exist in the context of the Optimizer.

We're not actually making a request here, just verifying that the URL has a valid format. Therefore `https://example.com` is faster and is easier to test, so I think this should be fine here.

Reply:
> `home_url()` doesn't exist in the context of the Optimizer.

That might have been unclear. It doesn't exist within the context of the Optimizer because the Optimizer library is framework-agnostic and doesn't know about WordPress. This should work just as well with any other PHP code base.

Reply:
Oh right, it's only for validation. 👍